packages/eslint-plugin/src/utils/ast-utils.ts

```diff
@@ -27,9 +27,43 @@ function tryReadBaseJson() {
  */
 export function getBarrelEntryPointByImportScope(
   importScope: string
-): string[] | null {
+): string[] {
+  const getOneOrNone = <T, R>(a: T[], f: (a: T[]) => R): [] | [T] | R => {
+    if (a.length > 1) {
+      return f(a);
+    }
+    return a as [] | [T];
+  };
+  const tryPaths = (
+    paths: Record<string, string[]>,
+    importScope: string
+  ): string[] => {
+    // TODO check and warn that the entries of paths[importScope] have no wildcards; that'd be user misconfiguration
+    if (paths[importScope]) return paths[importScope];
+    const WILDCARD_TS_SPECIAL_CASE = '*';
+    // accommodate wildcards (it's not glob) https://www.typescriptlang.org/docs/handbook/module-resolution.html#path-mapping
+    return getOneOrNone(
+      Object.entries(paths)
+        .map(([src, dests]) => {
+          const isWildcard = src.endsWith(WILDCARD_TS_SPECIAL_CASE);
```

Comment:
Again, please use `*` directly. It makes the code more readable.

Reply:
Inlined